cellengine/resources/gate.py

```diff
@@ -193,51 +96,197 @@ def create(
+                will be made to find the file by name. If zero or more than one file
+                exists with the name, an error will be thrown. Looking up files
+                by name is slower than using the ID, as this requires
+                additional requests to the server. If specified, do not specify
+                ``fcs_file_id``.
+            create_population (bool): Automatically create corresponding population.
 
-class PolygonGate(Gate):
-    """Basic concrete class for polygon gates"""
+        Returns:
+            A RectangleGate object.
 
+        Example:
+            ```python
+            experiment.create_rectangle_gate(x_channel="FSC-A", y_channel="FSC-W",
+            name="my gate", 12.502, 95.102, 1020, 32021.2)
+            cellengine.Gate.create_rectangle_gate(experiment_id, x_channel="FSC-A",
+            y_channel="FSC-W", name="my gate", x1=12.502, x2=95.102, y1=1020,
+            y2=32021.2, gid=global_gate.gid)
+            ```
+        """
+
+        fcs_file_id = parse_fcs_file_args(
+            experiment_id, tailored_per_file, fcs_file_id, fcs_file,
+        )
+
+        if label == []:
+            label = [numpy.mean([x1, x2]), numpy.mean([y1, y2])]
+        if not gid:
+            gid = generate_id()
+
+        model = {
+            "locked": locked,
+            "label": label,
+            "rectangle": {"x1": x1, "x2": x2, "y1": y1, "y2": y2},
+        }
+
+        return Gate(
+            id=None,
+            experiment_id=experiment_id,
+            fcs_file_id=fcs_file_id,
+            gid=gid,
+            parent_population_id=parent_population_id,
+            tailored_per_file=tailored_per_file,
+            type="RectangleGate",
+            x_channel=x_channel,
+            model=model,
+            y_channel=y_channel,
+            name=name,
+            # TODO: create_population
+        )  # type: ignore
+
+
+@define(repr=False)
+class PolygonGate(Gate):
     @classmethod
-    @doc_inherit(format_polygon_gate)
     def create(
         cls,
         experiment_id: str,
         x_channel: str,
         y_channel: str,
         name: str,
-        vertices: List[float],
-        label: List = [],
-        gid: str = None,
-        locked: bool = False,
-        parent_population_id: str = None,
-        parent_population: str = None,
-        tailored_per_file: bool = False,
-        fcs_file_id: str = None,
-        fcs_file: str = None,
-        create_population: bool = True,
+        vertices: List[List[float]],
+        label: Optional[List[float]] = [],
+        gid: Optional[str] = None,
+        locked: Optional[bool] = False,
+        parent_population_id: Optional[str] = None,
+        parent_population: Optional[str] = None,
+        tailored_per_file: Optional[bool] = False,
+        fcs_file_id: Optional[str] = None,
+        fcs_file: Optional[str] = None,
+        create_population: Optional[bool] = True,
```

Comment:
It looks like this is the only gate subclass that has these as `Optional`s. Not positive, but it [sounds like](https://stackoverflow.com/a/62732458/1218408) only args with `None` defaults should be `Optional`, and that should be consistent across the gate subclasses.